### ISSUE TITLE
Feature request: Keras CRF layer

### ISSUE THREAD

**Issue by ipoletaev** [tensorflow/tensorflow#10215]
It will be great if in the `contrib.keras` in the nearest future linear CRF layer (inside the Keras [wrapper](https://github.com/fchollet/keras/issues/4090)) will be added.

I think a lot of people will find this very useful.

**Comment by jart** [comment]
Another feature request for you @fchollet.

**Comment by ipoletaev** [comment]
Thanks!

**Comment by KiddoThe2B** [comment]
It seems that BI-LSTM-CRF are performing in SOTA levels for sequence labelling tasks for almost 2 years. It would be nice to have this functionality supported in Keras 2.

- "Neural Architectures for Named Entity Recognition" (https://arxiv.org/pdf/1603.01360.pdf)
- "End-to-end Sequence Labeling via Bi-directional LSTM-CNNs-CRF" (http://www.aclweb.org/anthology/P16-1101)

**Comment by yanghanxy** [comment]
Hoping to have this feature soon. 
Linear Chain CRF really helps a lot in tasks such as POS tagging, named entity recognition and slot extraction.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by tatatodd** [comment]
What do you think @fchollet?  If you agree with this, please add the "contributions welcome" label.

**Comment by zixia** [comment]
+1

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 117 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 131 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 146 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 161 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 176 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 191 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by cy89** [comment]
@fchollet I'll switch to "contributions welcome".

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by dynamicwebpaige** [comment]
Though this is outside the scope of `tf.keras`, a linear-chain CRF layer might be a welcome addition to [TensorFlow Addons](https://github.com/tensorflow/addons/tree/master/tensorflow_addons/layers).

Transferring the issue; @seanpmorgan for visibility.
